app/jobs/application_job.rb

```diff
@@ -20,8 +20,11 @@ class ApplicationJob < ActiveJob::Base
   before_enqueue do |job|
     throw(:abort) unless job.class.before_enqueue_lock(*job.arguments)
   end
-  around_perform do |job, block|
-    job.class.around_perform_lock(*job.arguments, &block)
+
+  # Overriding so that changes in ActiveModel object don't result in new lock (which they do when just calling to_s).
+  def self.lock(*args)
+    new_args = args.map { |arg| arg.class.method_defined?(:to_global_id) ? arg.to_global_id.to_s : arg }
+    "lock:#{self.class.name}-#{new_args}"
```

Comment:
What is `new_args` here? An array? I'm wondering what the lock string looks like. Not that it really matters, as long as it's functioning as expected.

Reply:
i believe `new_args` is the translated version of the original `args`, for use as the lock's key, subbing in global ID for the vanilla `to_s` representation, so that object mutation between queuing and completion of the job doesn't result in attempt to purge the wrong lock (by trying to purge using a key for which a lock was never created).

Reply:
`lock_args` or something might've been a slightly more precise name... top level method comment is sufficient imho, but maybe we should leave a bit more detail or use a slightly more precise name for our future selves?

Reply:
so `Array#to_s` is OK in this context (vs., say, `Array#join`)?

Reply:
i think so... a quick/simple test example to that effect makes the `#to_s` result seem reasonable in comparison to the `#join` result:
```ruby
[10] pry(main)> test_args = ['foo', 'bar', 'baz']
=> ["foo", "bar", "baz"]
[11] pry(main)> test_args.to_s
=> "[\"foo\", \"bar\", \"baz\"]"
[12] pry(main)> test_args.join
=> "foobarbaz"
```

i dunno if a person will ever be reading this value, but `#to_s` seems more human readable, so might even be preferable? 🤷‍♂

Reply:
because we don't really care about this value other than its uniqueness, and arrays are ordered so... agree.